gematria/datasets/find_accessed_addrs_from_bhive.cc

```diff
@@ -34,6 +39,47 @@ ABSL_FLAG(bool, quiet, false, "Omit all output except for the final summary");
 ABSL_FLAG(std::string, failing_blocks_csv, "",
           "Filename of an output CSV file to which any failing blocks are "
           "written. This can be used as an input for subsequent runs.");
+ABSL_FLAG(std::string, exegesis_snippets_dir, "",
+          "Directory to write EXEgesis snippets to");
+
+void WriteRegisterDef(std::ofstream& snippets_file, std::string_view name,
+                      std::optional<int64_t> value) {
+  if (!value) return;
+  snippets_file << "# LLVM-EXEGESIS-DEFREG " << name << " "
+                << absl::StrFormat("%016x", *value) << "\n";
+}
+
+void WriteExegesisSnippet(gematria::BHiveImporter& bhive_importer,
+                          std::string_view snippets_dir,
+                          const std::vector<uint8_t>& code,
+                          const gematria::AccessedAddrs& addrs, int n) {
+  auto proto = bhive_importer.BasicBlockProtoFromMachineCode(code);
```

Comment:
One more TODO to add (feel free to assign to me or Aiden): merge the support code in both versions of the accessed addresses code.

Reply:
It seems like `convert_bhive_to_llvm_exegesis_input.cc` is pretty similar to this script? I think we can do more than just merge the support code and probably just deprecate one or the other.

Reply:
Thanks for pointing me to that file. Yeah there's enough overlap that I think you're right. Looks like `convert_bhive_to_llvm_exegesis_input.cc` already emits these directives, but just uses hardcoded initial register values and memory values. Adding a flag to that tool to instead use `FindAccessedAddrs` to determine these values would make more sense than duplicating that here.

I initially created this tool just to have a way to test `FindAccessedAddrs` on a large dataset. If we do the above change then `convert_bhive_to_llvm_exegesis_input.cc` will serve that purpose perfectly well and `find_accessed_addrs_from_bhive.cc` can be removed. I'll add the TODO.